features/system-report/src/main/java/org/opennms/systemreport/event/CsvUtils.java

```diff
@@ -0,0 +1,196 @@
+        try (Reader reader = new FileReader(filePath, StandardCharsets.UTF_8);
+             CSVParser csvParser = new CSVParser(reader, CSVFormat.DEFAULT.withFirstRecordAsHeader())) {
+
+            String[] headers = csvParser.getHeaderMap().keySet().toArray(new String[0]);
+            try (Writer writer = new OutputStreamWriter(outputStream, StandardCharsets.UTF_8);
+                 CSVPrinter csvPrinter = new CSVPrinter(writer, CSVFormat.DEFAULT)) {
+
+                csvPrinter.printRecord((Object[]) headers);
+                csvPrinter.flush();
+            }
+        }
+
+        return new ByteArrayResource(outputStream.toByteArray());
+    }
+
+    public static Resource readCsvData(String filePath, boolean skipHeaders) throws IOException {
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        try (Reader reader = new FileReader(filePath, StandardCharsets.UTF_8);
+             CSVParser csvParser = new CSVParser(reader, CSVFormat.DEFAULT)) {
+            try (Writer writer = new OutputStreamWriter(outputStream, StandardCharsets.UTF_8);
+                 CSVPrinter csvPrinter = new CSVPrinter(writer, CSVFormat.DEFAULT)) {
+                boolean isFirstRow = true;
+                for (CSVRecord record : csvParser) {
+                    if (skipHeaders && isFirstRow) {
+                        isFirstRow = false;
+                        continue;
+                    }
+                    csvPrinter.printRecord(record);
+                }
+                csvPrinter.flush();
+            }
+        }
+        return new ByteArrayResource(outputStream.toByteArray());
+
+    }
+
+    public static void writeDataToFile(Resource csvHeaders, Resource csvData, String filePath) throws IOException {
+
+        File csvFile = new File(filePath);
+        boolean isNewFile = !csvFile.exists() && csvFile.createNewFile();
+        try (FileOutputStream fileOutputStream = new FileOutputStream(csvFile, true)) {
+            writeDataToStream(csvHeaders, csvData, fileOutputStream, isNewFile);
+        }
+    }
+
+    public static void writeDataToStream(Resource csvHeaders, Resource csvData, OutputStream outputStream,boolean writeHeaders) throws IOException {
+
+        CSVFormat csvFormat = CSVFormat.DEFAULT;
+        String[] headers = null;
+
+        if (writeHeaders && csvHeaders != null && csvHeaders.contentLength() > 0) {
+            try (InputStream headerStream = csvHeaders.getInputStream()) {
+                headers = StreamUtils.copyToString(headerStream, StandardCharsets.UTF_8).trim().split(",");
+                csvFormat = CSVFormat.DEFAULT.withHeader(headers);
+            }
+        }
+
+        //Writer and CSVPrinter are outside the try that prevents to close stream.
+        Writer writer = new BufferedWriter(new OutputStreamWriter(outputStream, StandardCharsets.UTF_8));
+        CSVPrinter csvPrinter = new CSVPrinter(writer, csvFormat);
```

Comment:
Not sure I understood this statement, why Writer and CSVPrinter are outside try block